subprojects/internal-testing/src/main/groovy/org/gradle/test/fixtures/Flaky.groovy

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.test.fixtures
+
+import org.junit.jupiter.api.Tag
+
+import java.lang.annotation.ElementType
+import java.lang.annotation.Retention
+import java.lang.annotation.RetentionPolicy
+import java.lang.annotation.Target
+
+/**
+ * Marks a test as flaky. The flaky tests are quarantined to run
```

Comment:
We need some documentation here how to handle
- Spock tests
- JUnit Jupiter tests
- JUnit 4 tests

Especially for JUnit 4 tests it seems like we need to do something differently. Though it would be great if we wouldn't have to.